libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -200,7 +201,7 @@ private void neverEntitled(Class<?> callerClass, Supplier<String> operationDescr
             return;
         }
 
-        throw new NotEntitledException(
+        notEntitled(
```

Comment:
I'm not the biggest fan of this change tbh; it does not make things more readable, and it adds a frame to the stack trace I think?
Not blocking, but I preferred the way it was.

Reply:
I did this so the throws are all done from one place in order to switch them all to reporting only in [this commit](https://github.com/elastic/elasticsearch/pull/122261/commits/7835c36e25d09a3b79711c99c45335a2aeecc403). That reporting mode was really useful; I suspect we'll end up making that "for real" somehow and merging it, but I didn't do that in this PR.

Reply:
I should say... in isolation, with no context, I agree with you and probably would have left the same review comment. 😂

Reply:
Yes! :) thanks for the context, the change makes sense now